### PR TITLE
gm: no testnet command, compatibility with multiple networks

### PR DESCRIPTION
**Pull request by greg-szabo**
Closes: #1050 

## Description
# Gaiad Manager Change Log

### FEATURES
- Compatibility of hermes updated to 0.4.1 and above. ([#1049])
- Enabled swagger page on the gaiad APP port.

### BUGFIXES
- Re-enable APP port in configuration ([comment](https://github.com/informalsystems/ibc-rs/pull/1051#issuecomment-856024919))

[#1049]: https://github.com/informalsystems/ibc-rs/issues/1049

### FEATURES
- Reorganized the documentation and moved the configuration file documentation into the example configuration.
- Added auto-configuration of the `denom` and `prefix` options in hermes config.

### BUGFIXES
- Fixed a small bug with the DEBUG features (used `==` instead of `=` when testing for the `DEBUG` flag.)
- Fixed the 5-network mesh example configuration in the documentation.
- Removed `--x-crisis-skip-assert-invariants` as not all networks support it.
- Only add node to the hermes config if there is no node already for that network

### REFACTORS
- Reorganized the `lib-gm` file to make it slightly easier to add configuration options.
- Simplified the empty "default" config that gets created if no config exists.
- Moved away from the `testnet` command as not all networks support it.
- Renamed `wallet_hdpath` configuration item to `hdpath` to reflect that the validator address is also impacted during
  creation.

______

For contributor use:

- [X] Updated the __Unreleased__ section of [CHANGELOG.md](https://github.com/informalsystems/ibc-rs/blob/master/CHANGELOG.md) with the issue.
- [X] If applicable: Unit tests written, added test to CI. - Tested on 5 different network binaries: gaiad, regen, akash, iris, persistence
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [X] Updated relevant documentation (`docs/`) and code comments.
- [X] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

**Comment by greg-szabo** [comment]
Note: enable APP port in app.toml (init keeps it false)

**Comment by greg-szabo** [comment]
With Romain's `hdpath` fix this morning `gm` needs another fix and bump the version to 0.0.6

File: scripts/gm/gm.toml [hunk 1]
```diff
@@ -1,38 +1,58 @@
...
-# except if otherwise noted.
+##
+## Global section defines application-wide defaults.
+##
+
+# All the entries here can be overridden on a per-node basis, except if indicated otherwise.
 [global]
 
-# Path to the `gaiad` binary to use.
+# Path to the `gaiad` (or other SDK-based) binary to use.
```
Comment (adizere) [inline scripts/gm/gm.toml:10]: Perhaps we can rename `gm` into `lcm` (local cosmos manager) or something like that later on ;p

File: scripts/gm/bin/lib-gm [hunk 2]
```diff
@@ -732,36 +760,39 @@ EOF
...
-denominator = '3'
+numerator='1'
+denominator='3'
 
 EOF
     done
 }
 
 # This is a helper function that extracts the coinType from an absolute hdpath (m/44'/118'/0'/0/0) for hermes.
+# This will be obsolete after https://github.com/informalsystems/ibc-rs/issues/1049 .
```
Comment (adizere) [inline scripts/gm/bin/lib-gm:790]: Is it obsolete now?

**Review by adizere — COMMENTED** [review]
Did not have any critical comments. I tested the usual workflows and looks good! Looking forward to 0.0.6.

File: scripts/gm/gm.toml (hunk 1 above)
Comment (greg-szabo) [reply scripts/gm/gm.toml:10]: Open to any naming. :) ("local cosmos", LOL) I kept it two letters for shortness's sake (and to adhere to age-old Unix preferences).

File: scripts/gm/bin/lib-gm (hunk 2 above)
Comment (greg-szabo) [reply scripts/gm/bin/lib-gm:790]: It will be obsolete in 0.0.6 and I'll remove it there.

File: scripts/gm/gm.toml (hunk 1 above)
Comment (adizere) [reply scripts/gm/gm.toml:10]: "loco" (spanish for crazy) also sounds fun.

**Review by adizere — APPROVED** [review]
Thank you Greg, LGTM!

**Pull request merged**